infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationMetadataManager.java

```diff
@@ -301,21 +313,85 @@ public void setSnapshotApplied(LogReplicationEntry entry) {
                 " persistSiteConfigID " + persistSiteConfigID + " persistSnapStart " + persistSnapStart);
     }
 
+    public void setReplicationRemainingPercent(long percentComplete) {
+        ReplicationStatusKey key = ReplicationStatusKey.newBuilder().setKey(
+                ReplicationStatusKey.Status.REPLICATION_COMPLETION).build();
+        ReplicationStatusVal val = ReplicationStatusVal.newBuilder().setReplicationCompletion(percentComplete).build();
+        TxBuilder txBuilder = corfuStore.tx(NAMESPACE);
+        txBuilder.update(REPLICATION_STATUS_TABLE, key, val, null);
+        txBuilder.commit();
+    }
+
+    public long getReplicationRemainingPercent() {
+        ReplicationStatusKey key = ReplicationStatusKey.newBuilder().setKey(
+                ReplicationStatusKey.Status.REPLICATION_COMPLETION).build();
+        CorfuRecord record = corfuStore.query(NAMESPACE).getRecord(REPLICATION_STATUS_TABLE, key);
+        if (record == null) {
+            return 0;
+        }
+        ReplicationStatusVal val = (ReplicationStatusVal)record.getPayload();
+        return val.getReplicationCompletion();
+    }
+
+    public void setDataConsistentOnStandby(boolean isConsistent) {
+        ReplicationStatusKey key = ReplicationStatusKey.newBuilder().setKey(
+                ReplicationStatusKey.Status.IS_DATA_CONSISTENT).build();
+        ReplicationStatusVal val = ReplicationStatusVal.newBuilder().setDataConsistent(isConsistent).build();
+        TxBuilder txBuilder = corfuStore.tx(NAMESPACE);
+        txBuilder.update(REPLICATION_STATUS_TABLE, key, val, null);
+        txBuilder.commit();
+    }
+
+    public boolean getDataConsistentOnStandby() {
+        ReplicationStatusKey key = ReplicationStatusKey.newBuilder().setKey(
+                ReplicationStatusKey.Status.IS_DATA_CONSISTENT).build();
+        CorfuRecord record = corfuStore.query(NAMESPACE).getRecord(REPLICATION_STATUS_TABLE, key);
+
+        // TODO: Initially, snapshot sync is pending so the data is not consistent.  But what if everything was applied and
+        // the standby rebooted?  In that case it is consistent
+        if (record == null) {
+            return false;
```

Comment:
Also here we can print some warning message or something for us to know its not yet persisted.